spyder/plugins/tests/test_ipythonconsole.py

```diff
@@ -323,6 +323,60 @@ def test_console_coloring(ipyconsole, qtbot):
     assert console_font_color.strip() == editor_font_color.strip()
 
 
+@pytest.mark.slow
+@flaky(max_runs=3)
+@pytest.mark.skipif(os.name == 'nt', reason="It times out sometimes on Windows")
+def test_set_cwd(ipyconsole, qtbot):
+    """Test kernel when changing cwd."""
+    # Wait until the window is fully up
+    shell = ipyconsole.get_current_shellwidget()
+    qtbot.waitUntil(lambda: shell._prompt_html is not None,
+                    timeout=SHELL_TIMEOUT)
+
+    # Issue 6451.
+    savetemp = shell._cwd
+    tempdir = tempfile.mkdtemp(suffix="queen's")
```

Comment:
Please use the `tmpdir` fixture instead of `tempfile`. There's an example of it here:

https://github.com/spyder-ide/spyder/blob/3.x/spyder/app/tests/test_mainwindow.py#L213

Reply:
Thanks, I've made the change.